grpc/src/js_client/tx_client.rs

```diff
@@ -0,0 +1,332 @@
+use celestia_proto::cosmos::tx::v1beta1::SignDoc;
+use celestia_types::any::JsAny;
+use celestia_types::consts::appconsts::JsAppVersion;
+use celestia_types::state::auth::{JsAuthParams, JsBaseAccount};
+use celestia_types::state::JsCoin;
+use celestia_types::Blob;
+use js_sys::{BigInt, Function, Promise, Uint8Array};
+use k256::ecdsa::signature::Error as SignatureError;
+use k256::ecdsa::{Signature, VerifyingKey};
+use lumina_utils::make_object;
+use prost::Message;
+use tonic_web_wasm_client::Client;
+use wasm_bindgen::prelude::*;
+use wasm_bindgen::JsCast;
+use wasm_bindgen_futures::JsFuture;
+
+use crate::tx::{DocSigner, JsTxConfig, JsTxInfo};
+use crate::{Result, TxClient};
+
+/// Celestia grpc transaction client.
+#[wasm_bindgen(js_name = "TxClient")]
+pub struct JsTxClient {
+    client: TxClient<Client, JsSigner>,
+}
+
+#[wasm_bindgen(js_class = "TxClient")]
+impl JsTxClient {
```

Comment:
I'm afraid it cannot be done without writing the actual boilerplate, but I think tx client should be a superset of grpc client also in wasm, so it should now get also all the methods that are present in the js version of grpc client

Reply:
also I noticed the grpc client uses mostly `&str`s and here we use mostly `String`s, the common client impl would make that consistent too